x/execution/internal/keeper/keeper.go

```diff
@@ -194,6 +194,24 @@ func (k *Keeper) Create(ctx sdk.Context, msg types.MsgCreate) (*executionpb.Exec
 		if err := exec.Execute(); err != nil {
 			return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, err.Error())
 		}
+
+		// emit event
+		event := sdk.NewEvent(
+			types.EventType,
+			sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeActionCreated),
+			sdk.NewAttribute(types.AttributeKeyHash, exec.Hash.String()),
+			sdk.NewAttribute(types.AttributeKeyAddress, exec.Address.String()),
+			sdk.NewAttribute(types.AttributeKeyStatus, exec.Status.String()),
+			sdk.NewAttribute(types.AttributeKeyExecutor, exec.ExecutorHash.String()),
+			sdk.NewAttribute(types.AttributeKeyInstance, exec.InstanceHash.String()),
+		)
+		if !exec.ProcessHash.IsZero() {
+			event = event.AppendAttributes(
+				sdk.NewAttribute(types.AttributeKeyExecutor, exec.ProcessHash.String()),
+			)
+		}
+		ctx.EventManager().EmitEvent(event)
```

Comment:
I could move this event out of the "inProgress" condition so it will emit this event everytime this keeper function is called. Like this we would have event also for proposed execution.
@antho1404 what do you think?

Reply:
I think it would be better to use the status of the execution for the `action` and yes always emit this event, when an execution is proposed, created, completed or failed

Reply:
I would also put that just after the `store.Set`, basically at the end of this keeper